### ISSUE TITLE
No push-notification show when app is closed

### ISSUE THREAD

**Issue by rlazom** [zo0r/react-native-push-notification#1329]
When my react native app is running, it show the notifications perfectly, even if the app is minimized.
But when the app is (killed/closed), the schedule push notifications are no longer working...

Can you provide some help??

**Comment by davidrenji** [comment]
Did you find a solution? I just get the pop up alert when the app is in foreground, otherwise it just shows the notification in the tray but there's not pop up.

**Comment by HoanHearti** [comment]
any update?

**Comment by Dallas62** [comment]
Hi!
Can you provide more information ? (Platform, version, ...)
Thanks!

**Comment by HoanHearti** [comment]
Hi @Dallas62 
Platform: android, iOS,
iOS: 13.4.1 and 12.4.1 
version: 3.1.9,
Can you help me? My app show the notifications perfectly on simulator android.

**Comment by Dallas62** [comment]
Does it appear on both Android and iOS ?

**Comment by rikur** [comment]
I just fixed the issue on my app by adding:

```
        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerServiceGcm"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
            </intent-filter>
        </service>
```

to my AndroidManifest.xml. I had it there before, but it was missing the "Gcm" from the name..

**Comment by rikur** [comment]
Local notifications are working with the app closed on Android 10. My Android 8 test device still doesn't show the notifications if the app is closed.

**Comment by rikur** [comment]
BTW @Dallas62 thank you for maintaining this project! The state of Notification libraries for react-native is really grim at the moment.

**Comment by HoanHearti** [comment]
@rikur  can you show me your AndroidManifest.xml and build.gradle file?  Please

**Comment by HoanHearti** [comment]
> Does it appear on both Android and iOS ?

My android device os Androis 8, don't show background notification. My ios device ios13,  don't show notification both foreground and background

**Comment by Dallas62** [comment]
Hi @HoanHearti 
For iOS, I suppose it's not from this library ( #1364 )
But  `@react-native-community/push-notification-ios`take a look at it.

For Android, @rikur and @HoanHearti, is it push notification or scheduled notification ?

**Comment by rikur** [comment]
a local scheduled notification.

**Comment by Dallas62** [comment]
@rikur It's strange you add to  add GCM to make it work, did you tried with:
```xml

        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>
```
before this change ?

**Comment by rlazom** [comment]
For me it works on Android, it does when app is in foreground or closed, but is NOT if I kill the process. 
Is there any way or option to achieve this??

To, let's say, if you want keep your alarm no matter what. 
Like Android OS Alarm Service??

**Comment by Dallas62** [comment]
Hi @rlazom 
I just published a new version with the readme updated,
Can you check the new release?

**Comment by Dallas62** [comment]
Hi, I think i found what is happening, the channel is not create by the library, so remote notification are not handle by the application.
I will look into it ASAP.

**Comment by Dallas62** [comment]
Can you test the version on `master` ?

**Comment by ab5y** [comment]
> 
> 
> @rikur It's strange you add to add GCM to make it work, did you tried with:
> 
> ```
>         <service
>             android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
>             android:exported="false" >
>             <intent-filter>
>                 <action android:name="com.google.firebase.MESSAGING_EVENT" />
>             </intent-filter>
>         </service>
> ```
> 
> before this change ?

I am having the same issue of notifications not working when the app is in background. Wrt to the above Manifest file setup, Android studio doesn't seem to recognize any of the following paths:
`com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher`
`com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver`
`com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService`
`com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService`

![image](https://user-images.githubusercontent.com/4251453/80581200-725a4700-8a2a-11ea-906a-220c28325e6e.png)

The rest of the directions seem to work fine, and notifications do reach the app when it is in the foreground. I am guessing the above Manifest file issue and the background notifications malfunctioning issues are related?

**Comment by Dallas62** [comment]
Hi @ab5y 
You can test the version on master, I will release a version today to fix this. A default channel is missing.

**Comment by Dallas62** [comment]
Seems to be fix in `3.3.1`.
